Code/LMEM_and_Plots.Rmd

```diff
@@ -0,0 +1,372 @@
+library(sjlabelled)
+
+#Outputs a cleaned up version of the model.
+#tab_model(mixed_model_interactions_no_PP)
+
+```
+
+###PLOTTING
+
+```{r}
+#Predator mass by lifestage: as they get bigger, they get older
+ggplot(fish_filter, aes(x=Predator.lifestage, y=logpredmass))+
+  geom_boxplot()+
+  theme_bw()+scale_x_discrete(limits=c("Larva","Juvenile","Adult"))+
+  labs(y="Log (Predator mass)", x="Predator Lifestage")
+                                                           
+#ggsave("Lifestage_LogPredMass.jpg", plot = last_plot(), device = NULL, path = NULL,
+#  scale = 1, width = NA, height = NA, units = c("in", "cm", "mm"),
+#  dpi = 300, limitsize = TRUE)
+
+```
+
+PLOTTING PRIMARY PRODUCTIVITY
+```{r}
+#Primary productivity is removed from the model. This is not unreasonable, given that there
+#is no obvious relationship when you plot the points: 
+#We do see that the shelf habitat shows considerable variation in 
+#predator mass for same mean primary productivity, 
+#suggesting a greater diversity in that environment. This relationship just isn't linear.
+
+library(scales)
+
+ggplot(fish_filter, aes(x=Mean_PP_Standardized, y=log10(Predator.mass), 
+                        colour=Specific.habitat))+
+  geom_boxplot(width=0.8)+theme_bw()+
+  labs(x="Mean Primary Productivity (Standardized)", 
+       y="log (Predator mass)", fill="Geographic Location")+
+  scale_x_continuous(labels=comma)
+
+#ggsave("MPP_LogPredMass.jpg", plot = last_plot(), device = NULL, path = NULL,
+#  scale = 1, width = NA, height = NA, units = c("in", "cm", "mm"),
+#  dpi = 300, limitsize = TRUE)
+
+```
+PLOTTING PREY MASS
+
+```{r}
+#Prey mass significantly and positively predicts predator mass, but weakly.
+#This is reflected by the low coefficient of our model
+#Trend is primarily driven by Atlantic ocean fish
+#Removed outliers to make trend more obvious
+#Most slopes are positive
+
+ggplot(fish_filter, aes(x=Prey.mass.standardized,
+                        y=logpredmass, colour=Geographic.location))+
+  geom_point(alpha=0.3)+xlim(c(0,34))+ylim(0,6)+
+  theme_classic()+
+  labs(x="Prey mass (standardized)", y="Predator mass (log)")+
+  geom_smooth(method="lm", se=F)
+
+# ggsave("Prey_LogPredMass.jpg", plot = last_plot(), device = NULL, path = NULL,
+#   scale = 1, width = NA, height = NA, units = c("in", "cm", "mm"),
+#  dpi = 300, limitsize = TRUE)
+
+```
+
+PLOTTING FEEDING INTERACTION 
+
+```{r}
+
+ ggplot(fish_filter,aes(x=Type.of.feeding.interaction, y=logpredmass)) +
+  geom_boxplot()+
+  theme_classic() +labs(y="Log (Predator mass)",
+                        x="Predator Feeding Interaction")+
+                        scale_x_discrete(limits=c("Planktivorous","Piscivorous","Predacious"))         
+                                           
+
+# ggsave("FeedingInteraction_LogPredMass.jpg", plot = last_plot(), 
+#device = NULL, path = NULL,
+#   scale = 1, width = NA, height = NA, units = c("in", "cm", "mm"),
+#   dpi = 300, limitsize = TRUE)
+
+

```

Comment:
# PLOTTING INTERACTION BETWEEN LIFE STAGE : FEEDING INTERACTION
fish_filter %>% 
  filter(Predator.lifestage != "Larva", Type.of.feeding.interaction != "planktivorous") %>% 
  group_by(Predator.lifestage, Type.of.feeding.interaction) %>% 
  summarise(meanlogpredmass = mean(logpredmass), n_obs = n(), 
            se_mass = (meanlogpredmass/sqrt(n_obs))) %>% 
  ggplot()+
  aes(x = Predator.lifestage, y = meanlogpredmass, colour = Type.of.feeding.interaction)+
  geom_line(aes(group=Type.of.feeding.interaction))+
  geom_errorbar(aes(ymin = (meanlogpredmass-se_mass), ymax = (meanlogpredmass + se_mass)),
                    width = 0.05)+
  geom_point()+
  labs(x = "Predator Lifestage", y = "Mean of Log Predator Mass", 
       colour = "Type of Feeding Interaction")+
  scale_x_discrete(limits = c("Juvenile", "Adult"))+
  scale_colour_manual(values = c("midnightblue", "orangered"))+
  theme_classic()